tests/unit/test_operator.py

```diff
@@ -77,10 +67,152 @@ def test_with_relation(harness):
+        harness_with_profiles.container_pebble_ready(CHARM_NAME)
+        harness_with_profiles.begin_with_initial_hooks()
+        assert (
+            harness_with_profiles.get_container_pebble_plan(CHARM_NAME)._services
+            is not None
+        )
+        assert isinstance(harness_with_profiles.charm.model.unit.status, ActiveStatus)
+
+    @patch("charm.KubeflowDashboardOperator.k8s_resource_handler", MagicMock())
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    @patch("charm.KubeflowDashboardOperator.container")
+    def test_update_layer_failure(
+        self,
+        container: MagicMock,
+        harness_with_profiles: Harness,
+    ):
+        container.replan.side_effect = _FakeChangeError(
+            "Fake problem during layer update", None
+        )
+        harness_with_profiles.container_pebble_ready(CHARM_NAME)
+        harness_with_profiles.begin_with_initial_hooks()
+        assert harness_with_profiles.charm.model.unit.status == BlockedStatus(
+            "Failed to replan"
+        )
+
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    @patch("charm.KubeflowDashboardOperator.k8s_resource_handler")
+    def test_create_resources_success(
+        self,
+        k8s_resource_handler: MagicMock,
+        harness_with_profiles: Harness,
+    ):
+        harness_with_profiles.begin()
+        harness_with_profiles.charm.on.install.emit()
+        k8s_resource_handler.apply.assert_called_once()
+        assert isinstance(harness_with_profiles.charm.model.unit.status, ActiveStatus)
+
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    @patch("charm.KubeflowDashboardOperator.k8s_resource_handler")
+    @patch("charm.KubeflowDashboardOperator._update_layer")
+    def test_main(
+        self,
+        update_layer: MagicMock,
+        k8s_resource_handler: MagicMock,
+        harness_with_profiles: Harness,
+    ):
+        expected_links = BASE_SIDEBAR
+        harness_with_profiles.begin()
+        harness_with_profiles.charm.on.install.emit()
+        k8s_resource_handler.apply.assert_called()
+        update_layer.assert_called()
+        assert isinstance(harness_with_profiles.charm.model.unit.status, ActiveStatus)
+        assert harness_with_profiles.charm._context["links"] == expected_links
+
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    @patch("charm.KubeflowDashboardOperator.k8s_resource_handler")
+    @patch("charm.delete_many")
+    def test_on_remove_success(
+        self,
+        delete_many: MagicMock,
+        k8s_resource_handler: MagicMock,
+        harness_with_profiles: Harness,
+    ):
+        harness_with_profiles.begin()
+        harness_with_profiles.charm.on.remove.emit()
+        k8s_resource_handler.assert_has_calls([mock.call.render_manifests()])
+        delete_many.assert_called()
+        assert isinstance(harness_with_profiles.charm.model.unit.status, ActiveStatus)
```

Comment:
No need (not possible also) to check the status after the charm has been removed

Reply:
So after discussion with pietro, he mentioned like the body of that event is some code which may end in failure which should be reflected in the status and that should be the way how to test it ... but not sure now

Reply:
The thing is that charm is not removed until the even successfully passes